### ISSUE TITLE
Flutter 3 code generation failed "This requires the 'super-parameters' language feature to be enabled."

### ISSUE THREAD

**Issue by Turbozanik** [Milad-Akarie/auto_route_library#1087]
Hi @Milad-Akarie,

I am using new super.X params.
And when I am trying to build my project after SDK update, I am receiving the following errors:

```
...
...
...
However, package:quiz_time/base/widgets/custom_toolbar.dart (or an existing part) contains the following errors.
custom_toolbar.dart:11:43: This requires the 'super-parameters' language feature to be enabled.

Try fixing the errors and re-running the build.

[INFO] Running build completed, took 18.0s

[INFO] Caching finalized dependency graph...
[INFO] Caching finalized dependency graph completed, took 47ms

[SEVERE] Failed after 18.0s
pub finished with exit code 1
```

My project sdk is  `sdk: ">=2.17.0 <3.0.0"`
And generator sdk is  `sdk: ">=2.12.0 <3.0.0"`

It seems like, that to fix this, all that has to be done, is to update dart sdk in https://github.com/Milad-Akarie/auto_route_library/blob/master/auto_route_generator/pubspec.yaml

I wanted to know, will this issue be fixed, or should I rollback to a lower flutter/auto_route version?

**Comment by Milad-Akarie** [comment]
@Turbozanik it seems to be the packing causing the error is not auto_route?

**Comment by Turbozanik** [comment]
@Milad-Akarie,
I think there is a misunderstanding here. 
Auto_route and auto_route_generator were litteraly the same lib for me. 
Technically it is auto_route_generator that is causing this issue.

1) I have upgraded my sdk to 2.17.0
2) I have used super-parameters feature.
3) And now, when I am trying to run `flutter pub run build_runner build --delete-conflicting-outputs`
    I am receiving the error that I have showed above.

package:quiz_time is my actual app.

As I understand(and I think I am right) auto_route_generator is dependent of SDK 2.12.0 and does not support this new super-parameters feature. 
And this is the root of the issue. As it has no idea about this new feature, it can not do the build.

And it seems like it will be fixed if we will set SDK to 2.17.0 here: https://github.com/Milad-Akarie/auto_route_library/blob/master/auto_route_generator/pubspec.yaml SDK
